### PR TITLE
fix(frontend): if you disallow cookies the page just shows errors

### LINKED ISSUE #1737: 🐛 [Bug] If you disallow cookies the page just shows errors
## 🐛 Bug

If you disallow cookies the page just shows errors:
image

After enabling cookies, I can signin.

### PR DESCRIPTION
**Pull request by trinity2701**
### Issues

- fixes #1737
- None

### Todo

- [X] None

### CODE REVIEW

File: frontend/renderer/app.ts [hunk 1]
```diff
@@ -27,7 +27,17 @@ function createApp(pageContext: PageContext, isClient = true) {
         DefaultApolloClient,
         createApolloClient(pageContext.publicEnv.ENDPOINTS)(getToken, isClient),
       )
-      provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      try {
+        provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      } catch {
+        if (navigator.cookieEnabled) {
+          toast.error('🍪 Please enabled cookies!', {
```
Comment (roschaefer) [inline frontend/renderer/app.ts:34]:
```suggestion
          toast.error('🍪 Please enable cookies!', {
```

File: frontend/renderer/app.ts [hunk 2]
```diff
@@ -27,7 +27,17 @@ function createApp(pageContext: PageContext, isClient = true) {
         DefaultApolloClient,
         createApolloClient(pageContext.publicEnv.ENDPOINTS)(getToken, isClient),
       )
-      provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      try {
+        provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      } catch {
```
Comment (roschaefer) [inline frontend/renderer/app.ts:32]: It's never a great idea to catch-all errors, we even have a ticket for this: https://github.com/dreammall-earth/dreammall.earth/issues/1548

Could you check what's the error type when you disallow cookies?

File: frontend/renderer/app.ts (hunk 1 above)
Comment (roschaefer) [reply frontend/renderer/app.ts:34]: It would be nice to translate this string.

Comment (roschaefer) [reply frontend/renderer/app.ts:34]: And maybe it's better to write 
```suggestion
          toast.error('🍪 Please enable cookies in your browser!', {
```

**Review by roschaefer — CHANGES_REQUESTED** [review]
cookies

File: frontend/renderer/app.ts [hunk 3]
```diff
@@ -27,7 +27,17 @@ function createApp(pageContext: PageContext, isClient = true) {
         DefaultApolloClient,
         createApolloClient(pageContext.publicEnv.ENDPOINTS)(getToken, isClient),
       )
-      provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      try {
+        provide('authService', new AuthService(pageContext.publicEnv.AUTH))
+      } catch (error) {
+        if (String(error).includes('SecurityError: Failed to read')) {
```
Comment (roschaefer) [inline frontend/renderer/app.ts:33]: This is what I see on firefox (Mozilla Firefox 130.0.1):

satty-20240927-13:01:22

**Review by roschaefer — CHANGES_REQUESTED** [review]
On chromium I'm not even able to disallow *all* cookies for any site. @trinity2701  How did you test it out?

File: frontend/renderer/app.ts [hunk 4]
```diff
@@ -27,7 +27,17 @@ function createApp(pageContext: PageContext, isClient = true) {
...
+          toast.error(i18n.global.t('error.enablecookie'), {
+            autoClose: false,
+            closeButton: false,
+            closeOnClick: false,
+          })
+        }
+      }
```
Comment (roschaefer) [inline frontend/renderer/app.ts:40]:
```suggestion
      try {
        provide('authService', new AuthService(pageContext.publicEnv.AUTH))
      } catch (error) {
        if (!error instancof WellKnownErrorClass) {
          throw error
        }
        toast.error(i18n.global.t('error.enablecookie'), {
          autoClose: false,
          closeButton: false,
          closeOnClick: false,
        })
      }
```

You also need to re-throw the error, if it's not the expected error!

**Comment by ulfgebhardt** [comment]
> On chromium I'm not even able to disallow _all_ cookies for any site. @trinity2701 How did you test it out?

Yes you can do this.

To test this, login, disallow cookies from `app....` and delete existing cookie. This simulales the behaviour

File: frontend/renderer/app.ts (hunk 4 above)
Comment (ulfgebhardt) [reply frontend/renderer/app.ts:40]: @roschaefer I think here is the problem we cannot find the class of `Security Error` - thats some browser related stuff

Comment (roschaefer) [reply frontend/renderer/app.ts:40]: It's OK! Then use `String(error).includes('SecurityError: Failed to read')` I just guess it's different for different browsers.

Important is re-throwing the error if the condition doesn't match. It's a shortcoming of JavaScript that `catch(err)` does not filter for error classes and this is an eternal source of bugs.

**Review by roschaefer — APPROVED** [review]
@trinity2701 said to me on chat that she's not available today, so I quickly fixed the issues with multiple browsers in 7458ed53079f668e2c75890a20aaf4ac87f668cf

File: frontend/renderer/app.ts (hunk 4 above)
Comment (roschaefer) [reply frontend/renderer/app.ts]: @ulfgebhardt The error is documented here: https://developer.mozilla.org/en-US/docs/Web/API/DOMException#securityerror

**Pull request merged**